pkg/remote/kyma_synchronization_context.go

```diff
@@ -141,24 +164,52 @@ func (c *KymaSynchronizationContext) CreateOrUpdateCRD(ctx context.Context, plur
 	)
 
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
-	err = c.RuntimeClient.Get(
+	err = runtimeClient.Get(
 		ctx, client.ObjectKey{
 			Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group),
 		}, crdFromRuntime,
 	)
 
-	if k8serrors.IsNotFound(err) || !ContainsLatestVersion(crdFromRuntime, v1beta2.GroupVersion.Version) {
-		return PatchCRD(ctx, c.RuntimeClient, crd)
+	kcpAnnotation := v1beta2.KcpKymaCRDGenerationAnnotation
+	skrAnnotation := v1beta2.SkrKymaCRDGenerationAnnotation
+
+	if plural == v1beta2.ModuleTemplateKind.Plural() {
+		kcpAnnotation = v1beta2.KcpModuleTemplateCRDGenerationAnnotation
+		skrAnnotation = v1beta2.SkrModuleTemplateCRDGenerationAnnotation
+	}
+
+	latestGeneration := strconv.FormatInt(crd.Generation, 10)
+	runtimeCRDGeneration := strconv.FormatInt(crdFromRuntime.Generation, 10)
+	if k8serrors.IsNotFound(err) || !ContainsLatestVersion(crdFromRuntime, v1beta2.GroupVersion.Version) ||
```

Comment:
This if condition you can extract to a function, and design a unit test for it, basically this is the most important part for this feature request. You test in what kind of situation, we are patching remote CRD.